test/AutoDiff/Sema/derivative_attr_type_checking.swift

```diff
@@ -185,6 +185,11 @@ protocol StaticMethod: Differentiable {
   static func generic<T: Differentiable>(_ x: T) -> T
 }
 
```

Comment:
I feel we need an error message stating clearly that defining derivatives for protocol requirements is not yet supported.

Reply:
Customized diagnostics may require extra support in `findAbstractFunctionDecl` in `TypeCheckAttr.cpp`. Currently, when `std::function<bool(AbstractFunctionDecl *)> &isValidCandidate` returns `false` for all candidates, a single generic error is emitted (`std::function<void()> &noneValidDiagnostic`).

We could change `isValidCandidate` emit diagnostics as a side effect, but this requires clever diagnostic cancellation (via `DiagnosticTransaction`) in case a valid candidate is found.

Alternatively, we could keep track of all invalid candidates and change `noneValidDiagnostic` to `std::function<void(AbstractFunctionDecl *)> &notValidDiagnostic`, so customized diagnostics can be emitted for every invalid candidate.

Reply:
This seems like more work than it's worth, especially since it's a temporary limitation, so I won't do this in this PR.